pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -61,15 +61,35 @@ func (pl *TaintToleration) Filter(ctx context.Context, state *framework.CycleSta
 	if err != nil {
 		return framework.NewStatus(framework.Error, err.Error())
 	}
+	taintsNotTolerated := []v1.Taint{}
 
-	if v1helper.TolerationsTolerateTaintsWithFilter(pod.Spec.Tolerations, taints, func(t *v1.Taint) bool {
+	if taintsNotTolerated = v1helper.TaintsNotToleratedWithFilter(taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
 		// PodToleratesNodeTaints is only interested in NoSchedule and NoExecute taints.
 		return t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectNoExecute
-	}) {
+	}); len(taintsNotTolerated) == 0 {
 		return nil
 	}
 
-	return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonNotMatch)
+	errReason := getTolerationErrReason(taintsNotTolerated, pod.Spec.Tolerations)
+	return framework.NewStatus(framework.UnschedulableAndUnresolvable, errReason)
+}
+
+func getTolerationErrReason(taintsNotTolerated []v1.Taint, tolerations []v1.Toleration) string {
```

Comment:
error reason should be simpler than that, something like: ```1 node(s) had taint xyz that the pod didn't tolerate.```

Reply:
right, I would include the value as well.

Reply:
updated the message